modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/tasks/SchedulerChannelProvider.java

```diff
@@ -0,0 +1,179 @@
+package org.motechproject.scheduler.tasks;
+
+import org.motechproject.commons.sql.util.Drivers;
+import org.motechproject.scheduler.constants.SchedulerConstants;
+import org.motechproject.tasks.domain.DynamicChannelProvider;
+import org.motechproject.tasks.domain.TriggerEvent;
+import org.motechproject.tasks.ex.TriggerRetrievalException;
+import org.quartz.JobDataMap;
+import org.quartz.utils.DBConnectionManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+@Service
+public class SchedulerChannelProvider implements DynamicChannelProvider {
```

Comment:
We should mark the task dependency as optional, similar to what commcare and other modules do